qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/model.py

```diff
@@ -0,0 +1,572 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import os
+import subprocess
+import shutil
+import time
+import datetime
+import json
+
+import matplotlib.pyplot as plt
+import numpy as np
+from mantid.geometry import CrystalStructure, ReflectionGenerator, ReflectionConditionFilter
+from mantid.simpleapi import CreateWorkspace, LoadGSS, DeleteWorkspace, CreateEmptyTableWorkspace, logger
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.tabs.gsas2 import parse_inputs
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.tabs.common import output_settings
+from mantidqtinterfaces.Engineering.gui.engineering_diffraction.settings.settings_helper import get_setting
+
+
+class GSAS2Model(object):
+
+    def __init__(self):
+        self.user_save_directory = None
+        self.project_name = None
+        self.data_files = None
+        self.instrument_files = None
+        self.x_min = None
+        self.x_max = None
+        self.data_x_min = None
+        self.data_x_max = None
+        self.number_of_regions = None
+        self.number_histograms = None
+        self.refinement_method = None
+        self.dSpacing_min = 1.0
+        self.timeout = 10
+
+    def clear_input_components(self):
+        self.user_save_directory = None
+        self.project_name = None
+        self.data_files = None
+        self.instrument_files = None
+        self.x_min = None
+        self.x_max = None
+        self.data_x_min = None
+        self.data_x_max = None
+        self.number_of_regions = None
+        self.number_histograms = None
+        self.refinement_method = None
+        self.dSpacing_min = 1.0
+        self.timeout = 10
+
+    def call_subprocess(self, command_string):
+        try:
+            shell_process = subprocess.Popen([command_string.replace('"', '\\"')],
```

Comment:
For me on Windows the presence of the `[` and `]` characters around the command was causing the call to GSAS to fail with this error:
`The filename, directory name, or volume label syntax is incorrect.`
If I remove the square brackets this problem goes away. Looking at the documentation for the python subprocess class (https://docs.python.org/3/library/subprocess.html) I think this is perhaps a Windows thing because there's some Windows specific handling when the args parameters is a list. Having said that perhaps the command_string variable isn't a comma separated list that would become a python list when wrapping in square brackets. So maybe the square brackets could be dropped without any downside on all platforms

Reply:
I thought I'd seen advice somewhere that said to use a list with Popen, but it does work without!